### PR TITLE
VTCCA-3017: play 2.6 + latest bootstrap + play-ui

### PR DESCRIPTION
**Pull request by PeterPerhac**

### CODE REVIEW

**Comment by darrell-plant** [comment]
LGTDz 👍

File: app/auth/GovernmentGatewayProvider.scala [hunk 1]
```diff
@@ -21,11 +21,14 @@ import javax.inject.Inject
...
-import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
 import scala.concurrent.Future
 
-class GovernmentGatewayProvider @Inject()(override val env: Environment, override val config: Configuration)(applicationConfig: ApplicationConfig) extends AuthRedirects {
+class GovernmentGatewayProvider @Inject()(
+                                           override val env: Environment,
+                                           override val config: Configuration
+                                         )(applicationConfig: ApplicationConfig) extends AuthRedirects {
```
Comment (DanielConnelly) [inline app/auth/GovernmentGatewayProvider.scala]: what are we using applicationConfig for, I ask as I want to remove these holder objects.

File: app/config/ApplicationConfig.scala [hunk 2]
```diff
@@ -78,15 +74,14 @@ class ApplicationConfig @Inject()(override val runModeConfiguration: Configurati
...
+  lazy val bannerContent: Option[String] = configuration.getOptional[String]("encodedBannerContent").map(
     e => new String(Base64.getUrlDecoder.decode(e)))
 
-  lazy val plannedImprovementsContent: Option[String] = runModeConfiguration.getString("plannedImprovementsContent").map(e =>
+  lazy val plannedImprovementsContent: Option[String] = configuration.getOptional[String]("plannedImprovementsContent").map(e =>
     new String(Base64.getUrlDecoder.decode(e)))
 
-  override protected def mode: Mode = environment.mode
-}
+  lazy val baseUrl: String = if (Set("Dev", "Test").contains(runMode.env)) "http://localhost:9523" else ""
```
Comment (DanielConnelly) [inline app/config/ApplicationConfig.scala]: what is this used for?

File: app/connectors/Addresses.scala [hunk 3]
```diff
@@ -19,17 +19,16 @@ package connectors
...
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
-import uk.gov.hmrc.play.config.ServicesConfig
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class Addresses @Inject()(config: ServicesConfig, http: HttpClient)(implicit executionContext: ExecutionContext) {
 
-  val url = config.baseUrl("property-linking") + "/property-linking/address"
+  lazy val url = config.baseUrl("property-linking") + "/property-linking/address"
```
Comment (DanielConnelly) [inline app/connectors/Addresses.scala]: why is this lazy.

File: app/auth/GovernmentGatewayProvider.scala (hunk 1 above)
Comment (PeterPerhac) [reply app/auth/GovernmentGatewayProvider.scala]: i just reformatted the file... it's getting current app's baseUrl from it

`Future.successful(toGGLogin(applicationConfig.baseUrl + request.uri))`

File: app/config/ApplicationConfig.scala (hunk 2 above)
Comment (PeterPerhac) [reply app/config/ApplicationConfig.scala]: it was there some somewhere higher up or in another class... 

redirects to localhost when run in test/dev mode... used in 5 places
image

File: app/connectors/Addresses.scala (hunk 3 above)
Comment (PeterPerhac) [reply app/connectors/Addresses.scala]: i think that's how i found it, made all of them un-lazy now :-D

**Comment by habeeb01** [comment]
LGTM 👍  (not sure if we should make sure Dan is happy before the merge ¯\_(ツ)_/¯ )

File: app/controllers/AddressLookup.scala [hunk 4]
```diff
@@ -16,29 +16,27 @@
...
   def findByPostcode(postcode: String): Action[AnyContent] = Action.async { implicit request =>
-      addresses.findByPostcode(postcode.trim)(hc) map { res =>
-        if (res.isEmpty) {
-          NotFound
-        } else {
-          Ok(Json.toJson(res))
-        }
-    } recover {
-        case _ => NotFound
+    addresses.findByPostcode(postcode.trim)(hc).recover { case _ => Seq.empty }.map {
```
Comment (DanielConnelly) [inline app/controllers/AddressLookup.scala]: can we log the exception here, I am not a fan of these catch all recovers without knowing what is causing these.

File: app/controllers/Application.scala [hunk 5]
```diff
@@ -36,9 +37,7 @@ class Application @Inject()(val errorHandler: CustomErrorHandler)(implicit val m
     Ok(views.html.start(RegisterHelper.choiceForm))
   }
 
-  def logOut() = Action { request =>
-    Redirect(routes.Application.start()).withNewSession
-  }
+  def logOut() = Action(Redirect(routes.Application.start()).withNewSession)
```
Comment (DanielConnelly) [inline app/controllers/Application.scala]: nice.

File: app/controllers/Register.scala [hunk 6]
```diff
@@ -21,10 +21,16 @@ import javax.inject.Inject
 import play.api.data.Form
 import play.api.data.Forms._
 import play.api.i18n.MessagesApi
-import play.api.mvc.{Action, AnyContent, Result}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
 import uk.gov.voa.propertylinking.errorhandler.CustomErrorHandler
 
-class Register @Inject()(override val errorHandler: CustomErrorHandler)(implicit override val messagesApi: MessagesApi, val config: ApplicationConfig) extends PropertyLinkingController {
+class Register @Inject()(
+                          override val errorHandler: CustomErrorHandler
```
Comment (DanielConnelly) [inline app/controllers/Register.scala]: as you made the errorHandler a def you dont need to override anymore.

File: app/controllers/detailedvaluationrequest/DvrController.scala [hunk 7]
```diff
@@ -193,22 +193,23 @@ class DvrController @Inject()(
...
         case Some(link) =>
           dvrCaseManagement
             .getDvrDocument(link.uarn, valuationId, link.submissionId, fileRef)
-            .map { document =>
-              Result(ResponseHeader(200, document.headers.updated(CONTENT_DISPOSITION, s"""attachment;filename="${link.submissionId}.pdf"""")),
-                     Streamed(document.body,
-                              document.contentLength,
-                              document.contentType))
+            .map { response =>
+              Ok.sendEntity(HttpEntity.Streamed(
```
Comment (DanielConnelly) [inline app/controllers/detailedvaluationrequest/DvrController.scala]: I like it!

Comment (DanielConnelly) [reply app/controllers/detailedvaluationrequest/DvrController.scala]: might steal that bit for something in dashboard-frontend

**Pull request merged**